Doc/library/urllib.parse.rst

```diff
@@ -37,30 +38,35 @@ URL Parsing
 The URL parsing functions focus on splitting a URL string into its components,
 or on combining URL components into a URL string.
 
-.. function:: urlparse(urlstring, scheme='', allow_fragments=True)
+.. function:: urlparse(urlstring, scheme='', allow_fragments=True, flags=SchemeFlag(0))
 
-   Parse a URL into six components, returning a 6-item :term:`named tuple`.  This
-   corresponds to the general structure of a URL:
-   ``scheme://netloc/path;parameters?query#fragment``.
-   Each tuple item is a string, possibly empty. The components are not broken up
-   into smaller parts (for example, the network location is a single string), and %
+   Parse a URL into six components with respect to given scheme classes,
+   returning a 6-item :term:`named tuple`. This corresponds to the general
+   structure of a URL: ``scheme://netloc/path;parameters?query#fragment``. Each
+   tuple item is a string, possibly empty. The components are not broken up into
+   smaller parts (for example, the network location is a single string), and %
    escapes are not expanded. The delimiters as shown above are not part of the
-   result, except for a leading slash in the *path* component, which is retained if
-   present.  For example:
+   result, except for a leading slash in the *path* component, which is retained
+   if present.
+
+   The scheme of the URL determines whether or not parameters are parsed as
+   distinct from the path. To override the scheme and parse parameters anyway,
+   pass the corresponding SchemeFlag.
+
+   For example:
 
    .. doctest::
       :options: +NORMALIZE_WHITESPACE
 
-      >>> from urllib.parse import urlparse
+      >>> from urllib.parse import urlparse, PARAMS
       >>> urlparse("scheme://netloc/path;parameters?query#fragment")
-      ParseResult(scheme='scheme', netloc='netloc', path='/path;parameters', params='',
-                  query='query', fragment='fragment')
+      ParseResult(scheme='scheme', netloc='netloc', path='/path;parameters', params='', query='query', fragment='fragment')
+      >>> urlparse("scheme://netloc/path;parameters?query#fragment", flags=PARAMS)
+      ParseResult(scheme='scheme', netloc='netloc', path='/path', params='parameters', query='query', fragment='fragment')
       >>> o = urlparse("http://docs.python.org:80/3/library/urllib.parse.html?"
       ...              "highlight=params#url-parsing")
       >>> o
-      ParseResult(scheme='http', netloc='docs.python.org:80',
-                  path='/3/library/urllib.parse.html', params='',
-                  query='highlight=params', fragment='url-parsing')
+      ParseResult(scheme='http', netloc='docs.python.org:80', path='/3/library/urllib.parse.html', params='', query='highlight=params', fragment='url-parsing')
```

Comment:
do not reformat doctest examples.  these were formatted to be narrow to avoid horizontal scrollbars in documentation on most common displays and to keep the .rst itself <80 columns when possible.

reformatting is unrelated to the change at hand and distracts from the actual change.

Reply:
Restoring the original formatting causes the doctest to fail, I should've broken that out into a separate and clear commit... I'm doctesting with `.python -m doctest`. Is that wrong, or is there some other way I can keep the old linebreaks?

Reply:
@gpshead re `behavior_overrides` vs `flags`: aren't flags usually behavior overrides?  `ssl`, `socket`, `_pydecimal`, `_osx_support`, and `re` all use `flags`, while `doctest` uses `compileflags`, `_pyio` use `dec_flags`, and `subprocess` uses `creationflags`.

My first choice here would be a simple `flags`, and it should be easily understood that the flags given will modify the parsing behavior of `urlparse`.  If we wanted to be more precise we could call it `scheme_flags` since that is the part being modified -- but `behavior_overrides` is no less generic and much more verbose than `flags`.